compiler/rustc_mir/src/transform/inline/cycle.rs

```diff
@@ -0,0 +1,79 @@
+use rustc_data_structures::fx::FxHashSet;
+use rustc_hir::def::DefKind;
+use rustc_hir::def_id::LocalDefId;
+use rustc_middle::mir::TerminatorKind;
+use rustc_middle::ty::{self, InstanceDef, TyCtxt};
+use std::collections::VecDeque;
+
+crate fn mir_inliner_cycle(tcx: TyCtxt<'tcx>, root: LocalDefId) -> bool {
+    let mut queue = VecDeque::new();
+    let mut seen = FxHashSet::default();
```

Comment:
maybe it is possible to improve the perf if a MiniSet from https://github.com/rust-lang/rust/pull/72412 can be used when I did som test on rustc yesterday the largest set was 36 see more results below.
so if the MiniSet was backed by a 36 element big array no allocations hade been needed but maybe not that big differense as the most is of len 0.

seen.len times 
0 20456
1 1026
2 596
3 368
4 173
5 67
6 57
7 62
8 20
9 19
10 33
11-19 92
20-29 17
30-36 3

Reply:
whoa! Thanks for collecting these stats! So around 10% of the cases have more than zero elements, if we use a mini-set with 4 elements only around 1% of the cases need allocations, so that seems like an ok middle ground

Reply:
I removed the need for a set entirely, so this is not applicable anymore.